analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py

```diff
@@ -15,6 +16,10 @@
 # TODO: delete unused mics to avoid fill up memory.
 # Dict containing (device_session_id:last_uplink_mic). Here it will be saved last uplink messages' MIC 
 last_uplink_mic = {}
+# Dict structure that stores device ids as keys and estimated retransmissions as values.
+# Works like a cached version of retransmission counters stored in database (device_counters table)
+# In the future, maybe we will need to store the other counters in this same structure as well.
+devices_retransmissions = {}
```

Comment:
This dict should be cleaned sometimes. It only saves and int, so, I don't think it will fill the memory, but if the data is too old could generate some errors that we didn't take into account. You could copy the structure of CheckDuplicatedSession, or CheckSessionRegeneration.

Reply:
I will add a garbage collector to delete old data